### ISSUE TITLE
Idempotent symlinking

### ISSUE THREAD

**Issue by ThonyPrice** [ThonyPrice/dotfiles#23]
Symlinking should be idempotent for cases
1. Target exists but is **not** a symlink
2. Target exists and _is_ a symlink already
3. Target does not exist

**Comment by ThonyPrice** [comment]
Rather verbose but this pattern is the best I've found so far:
``` yml
- name: Shell | Fish stat config dir
  stat:
    path: ~/.config/fish
  register: link

- name: Shell | Fish rm config dir (if not symlinked)
  file:
    path: ~/.config/fish
    state: absent
  when: link.stat.isdir is defined and not link.stat.islnk

- name: Shell | Fish config symlinked
  file: path=$HOME/.config/fish
        src=$HOME/dotfiles/roles/workstation/files/fish
        state=link
        force=yes

```
Implemented for all symlinks in #25

**Comment by ThonyPrice** [comment]
Closed by #25
